plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java

```diff
@@ -778,6 +790,35 @@ public Object[][] unsupportedTimestampDataProvider()
         };
     }
 
+    @Test(dataProvider = "sessionZonesDataProvider")
+    public void testClickHouseDateTimeWithTimeZone(ZoneId sessionZone)
+    {
+        Session session = Session.builder(getSession())
+                .setTimeZoneKey(TimeZoneKey.getTimeZoneKey(sessionZone.getId()))
+                .build();
+
+        dateTimeWithTimeZoneTest(clickhouseDateTimeInputTypeFactory("datetime"), clickhouseDateTimeInputLiteralFactory(), trinoTimestampExpectedLiteralFactory())
+                .execute(getQueryRunner(), session, clickhouseCreateAndInsert("tpch.datetime_tz"));
```

Comment:
why only `clickhouseCreateAndInsert`? `trinoCreateAsSelect` is also useful since CTAS and INSERT follow different code paths.

Reply:
Since Trino's `TIMESTAMP(p) WITH TIME ZONE` does not know the `timezone`, we cannot create ClickHouse's `DateTime(timezone)` through Trino.